src/java/voldemort/utils/Utils.java

```diff
@@ -148,45 +148,60 @@ public static void rm(Iterable<File> files) {
     }
 
     /**
-     * Creates a symbolic link to an existing file. If the symlink already exists and
-     * already points to the intended destination, no changes are made to the file-system.
-     * If the symlink already exists but points to wrong destination, it is deleted first
-     * before being recreated.
-     * 
-     * @param filePath Path of the file for whom to create the symbolic link
-     * @param symLinkPath Path of the symbolic link
-     */
-    public static void symlink(String filePath, String symLinkPath) {
-        File file = new File(filePath);
-        File symLink = new File(symLinkPath);
-
+     * Checks if a symlink already exists and already points to the intended destination
+     *
+     * @param file The file the symbolic link should point to
+     * @param symLink The symbolic link which should already exist
+     * @return true if the symlink exists and points to the intended destination, false otherwise.
+     * @throws VoldemortException if the destination file does not exist or IOException occurs.
+     */
+    public static boolean symlinkExists(File file, File symLink) {
```

Comment:
If I understand this function, it has four possible results:

1. symlink exists and has the expected value: return true, caller does nothing and does not need to keep trying
2. symlink exists and has a different valid value: return false, caller tries to resync from the filesystem
3. symlink doesn't exist: throw exception, caller does nothing and (sensibly) keeps trying
4. actually exceptional condition (parent directory missing, permissions issues): throw exception, caller does nothing and (foolishly) keeps trying

My problem is that case 3 is not sufficiently distinguished.  It's not an exceptional condition.  During recovery it's a normal and expected condition that can last for hours but will eventually go away.  Lumping together cases 3 and 4 into an undistinguished exception that results in the caller blindly retrying without logging, will make it harder to diagnose problems like filesystem corruption or permissions issues.